### PR TITLE
Automatically convert TS examples to JS

### PR DESCRIPTION
**Pull request by djhi**
Done client side and only when needed.
I compared locally with the previous docs and didn't notice any difference even with CPU throttling x6 on slow 3G

## How to test

- Open AccordionForm.md
- Click on the AccordionForm.Panel section 
- Scroll down
- The first snippet is bi-lingual
image

### CODE REVIEW

**Comment by fzaninotto** [comment]
When I click on a language tab, the screen "jumps". Can we avoid that?

**Comment by fzaninotto** [comment]
Also, it seems the indentation differs between JS and TS. Can we make it consistent?

**Comment by djhi** [comment]
> Also, it seems the indentation differs between JS and TS. Can we make it consistent?

I'm afraid we have inconsistent indentation everywhere so we'd have to apply the same everywhere as well

**Review by slax57 — CHANGES_REQUESTED** [review]
Code seems good.

BUT, I saw 2 issues:
- Once you switch to TSX, **all** JSX code snippets disappear! (it can be seen in `AccordionForm.md`)
- New lines are not preserved when they are in the JSX code. Example in the code below taken from `Tutorial.md`:

```tsx
// in src/App.tsx
import { Admin, Resource, ListGuesser, EditGuesser, ShowGuesser } from 'react-admin';
import { dataProvider } from './dataProvider';

export const App = () => (
    <Admin dataProvider={dataProvider}>

    </Admin>
);
```

The new lines between `<Admin>` and `</Admin>` are not preserved.

File: docs/css/style-v17.css [hunk 1]
```diff
@@ -655,4 +658,20 @@ body.no-sidebar .sidenav-trigger {
...
+
+.language-switcher {
+    background-color: #f5f2f0;
+    padding: 0.5rem 1rem;
+    transition: all 0.2s ease-in-out;
+}
+
+.language-switcher.active {
+    background-color: #f5f2f0;
+}
```
[the remark this answers is not in the document]
Comment (djhi) [reply docs/css/style-v17.css:677]: As discussed with Francois, it's actually unnecessary. @fzaninotto Did I get that right?

File: docs/js/ra-doc-exec.js [hunk 2]
```diff
@@ -1,5 +1,173 @@
...
+        {
+            plugins: [prettierPlugins.babel],
+            parser: 'babel',
+        }
+    );
+
+    return jsCode;
+};
+
+const builtJSCodeBlocksFromTS = async () => {
```
[the remark this answers is not in the document]
Comment (djhi) [reply docs/js/ra-doc-exec.js:90]: Typo

**Review by slax57 — APPROVED** [review]

**Comment by slax57** [comment]
@fzaninotto I'll let you check one last time :wink:

**Review by fzaninotto — APPROVED** [review]

**Pull request merged**